File: src/main/java/com/sparetimedevs/questlog/userpassword/UserPasswordRepository.java
package com.sparetimedevs.questlog.userpassword;

import com.sparetimedevs.questlog.user.User;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

public interface UserPasswordRepository extends CrudRepository<UserPassword, Long> {

	Optional<UserPassword> findByUser(@Param("USER") User user);

	Optional<UserPassword> findByUserEmailAddress(@Param("USER_EMAIL_ADDRESS") String userEmailAddress);

}

File: src/main/java/com/sparetimedevs/questlog/userpassword/UserPasswordRepository.kt
package com.sparetimedevs.questlog.userpassword

import com.sparetimedevs.questlog.user.User
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import java.util.Optional

interface UserPasswordRepository : CrudRepository<UserPassword, Long> {

	fun findByUser(@Param("USER") user: User): Optional<UserPassword>

	fun findByUserEmailAddress(@Param("USER_EMAIL_ADDRESS") userEmailAddress: String): Optional<UserPassword>

}
